functorch/_src/aot_autograd.py

```diff
@@ -1934,27 +1935,30 @@ def functional_call(*args, **kwargs):
         aot_id=next(AOT_COUNTER),
     )
 
-    compiled_fn = None
+    full_args = []
+    full_args.extend(params_flat)
+    full_args.extend(args)
 
-    @wraps(functional_call)
-    def compiled_f(*args):
-        nonlocal compiled_fn
-        if compiled_fn is None:
-            compiled_fn = create_aot_dispatcher_function(
-                functional_call,
-                args,
-                aot_config,
-            )
-        return compiled_fn(args)
-
-    def forward(*args):
-        return compiled_f(
-            *params_flat,
-            *args,
-        )
+    compiled_fn = create_aot_dispatcher_function(
+        functional_call,
+        full_args,
+        aot_config,
+    )
 
+    # TODO: There is something deeply wrong here; compiled_fn running with
```

Comment:
I don't think `compiled_fn` actually returns the boxed calling convention (`compiled_fn` here should be the `autograd.Function` object, not the callable returned from the backend.

Reply:
It gets boxed up here 

```
        if needs_autograd:
            return make_boxed_func(
                aot_dispatch_autograd(flat_fn, fake_flat_tensor_args, aot_config)
            )
```